ci-operator/step-registry/openstack/provision/sriov-worker/openstack-provision-sriov-worker-commands.sh

```diff
@@ -49,29 +49,37 @@ then
 	source "${SHARED_DIR}/proxy-conf.sh"
 fi
 
-if ! openstack network show "${OPENSTACK_SRIOV_NETWORK}" >/dev/null 2>&1; then
-    echo "Network ${OPENSTACK_SRIOV_NETWORK} doesn't exist"
-    exit 1
-fi
-NETWORK_ID=$(openstack network show "${OPENSTACK_SRIOV_NETWORK}" -f value -c id)
-SUBNET_ID=$(openstack network show "${OPENSTACK_SRIOV_NETWORK}" -f json -c subnets | jq '.subnets[0]' | sed 's/"//g')
-
-oc_version=$(oc version -o json | jq -r '.openshiftVersion')
-if [[ "${oc_version}" != *"4.9"* && "${oc_version}" != *"4.10"* && "$CONFIG_DRIVE" != "true" ]]; then
-    CONFIG_DRIVE=false
+# If this file is present, we want to run the tests against an Hypershift HostedCluster
+# and therefore we want to load the KUBECONFIG from a specific path.
+# We'll skip the creation of a new MachineSet because the node would already be deployed at
+# this point.
+if test -f "${SHARED_DIR}/nested_kubeconfig"
+then
+        export KUBECONFIG="${SHARED_DIR}/nested_kubeconfig"
 else
-    CONFIG_DRIVE=true
-fi
-
-echo "Downloading current MachineSet for workers"
-WORKER_MACHINESET=$(oc get machinesets.machine.openshift.io -n openshift-machine-api | grep worker | awk '{print $1}')
-oc get machinesets.machine.openshift.io -n openshift-machine-api "${WORKER_MACHINESET}" -o json > "${SHARED_DIR}/original-worker-machineset.json"
-
-if [[ "${OPENSTACK_SRIOV_NETWORK}" == *"hwoffload"* ]]; then
-    PROFILE="\"profile\": {\"capabilities\": \"[switchdev]\"},"
-fi
-
-cat <<EOF > "${SHARED_DIR}/sriov_patch.json"
+	if ! openstack network show "${OPENSTACK_SRIOV_NETWORK}" >/dev/null 2>&1; then
+	    echo "Network ${OPENSTACK_SRIOV_NETWORK} doesn't exist"
+	    exit 1
+	fi
+	NETWORK_ID=$(openstack network show "${OPENSTACK_SRIOV_NETWORK}" -f value -c id)
+	SUBNET_ID=$(openstack network show "${OPENSTACK_SRIOV_NETWORK}" -f json -c subnets | jq '.subnets[0]' | sed 's/"//g')
+	
+	oc_version=$(oc version -o json | jq -r '.openshiftVersion')
+	if [[ "${oc_version}" != *"4.9"* && "${oc_version}" != *"4.10"* && "$CONFIG_DRIVE" != "true" ]]; then
+	    CONFIG_DRIVE=false
+	else
+	    CONFIG_DRIVE=true
+	fi
```

Comment:
Not really related, but seeing as you're reindenting them: are either of 4.9 or 4.10 still supported/tested? We can likely simplify (in a separate commit) assuming not

Reply:
yeah right some cleanup needs to happen. I'll take care of it

Reply:
Done in this PR.